sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,258 @@
+     */
+    private List<CallInfo> forbiddenMethods;
+
+    /**
+     * Sets name of the config file.
+     *
+     * @param fileName name of the forbidden methods config file.
+     */
+    public void setFile(String fileName) {
+        file = fileName;
+    }
+
+    /**
+     * Sets whether config file existence is optional.
+     *
+     * @param optional tells if config file existence is optional.
+     */
+    public void setOptional(boolean optional) {
+        this.optional = optional;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+
+        return new int[]{
+            TokenTypes.METHOD_CALL,
+            TokenTypes.LITERAL_NEW,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    protected void finishLocalSetup() throws CheckstyleException {
+        if (file != null) {
+            if (optional) {
+                if (configFileExists(file)) {
+                    forbiddenMethods = ForbidCertainMethodsLoader.loadConfigValue(file);
+                }
+                else {
+                    forbiddenMethods = new ArrayList<>();
+                }
+            }
+            else {
+                forbiddenMethods = ForbidCertainMethodsLoader.loadConfigValue(file);
+            }
+        }
+    }
+
+    /**
+     * Checks if config file with given fileName exists.
+     *
+     * @param fileName name of the config file.
+     * @return true if config file exists, otherwise false
+     */
+    private static boolean configFileExists(String fileName) {
+        boolean configFileExists = true;
+        InputStream sourceInput = null;
+        try {
+            final URI uriByFilename = CommonUtils.getUriByFilename(fileName);
+            final URL url = uriByFilename.toURL();
+            sourceInput = url.openStream();
+        }
+        catch (CheckstyleException | IOException ignored) {
+            configFileExists = false;
+        }
+        finally {
+            if (sourceInput != null) {
+                try {
+                    sourceInput.close();
+                }
+                catch (IOException ignored) {
+                    configFileExists = false;
+                }
+            }
+        }
+        return configFileExists;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL: {
+                final DetailAST dot = ast.getFirstChild();
+                final String methodName = dot.getLastChild().getText();
+                final int numberOfParameters = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, CallInfo.of(methodName, numberOfParameters));
+                break;
+            }
+            case TokenTypes.LITERAL_NEW: {
```

Comment:
Technically constructors are not methods.
I question if this should be a separate check.
@romani What are your thoughts?

Reply:
ideally that should be different. But this Check is not ideal by limitations of Checkstyle.
It might be better to keep name - make big note in Check that it could run on C-tors if configured.